src/fhirtypes/ElementDefinition.ts

```diff
@@ -602,6 +623,301 @@ export class ElementDefinition {
+    }
+    return true;
+  }
+
+  /**
+   * Fixes a boolean to this element.
+   * @see {@link fixValue}
+   * @param {boolean} value - the boolean value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {MismatchedTypeError} when the type of the ElementDefinition is not boolean
+   */
+  fixBoolean(value: boolean): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError(typeof value);
+    }
+    const type = this.type[0].code;
+    if (type === 'boolean' && this.checkIfFixable(value, this.fixedBoolean, type)) {
+      this.fixedBoolean = value;
+    } else {
+      throw new MismatchedTypeError('boolean', value, type);
+    }
+  }
+
+  /**
+   * Fixes a number to this element.
+   * @see {@link fixValue}
+   * @param {number} value - the number value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {MismatchedTypeError} when the value does not match the type of the ElementDefinition
+   */
+  fixNumber(value: number): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError(typeof value);
+    }
+    const type = this.type[0].code;
+    if (type === 'decimal' && this.checkIfFixable(value, this.fixedDecimal, type)) {
+      this.fixedDecimal = value;
+    } else if (
+      type === 'integer' &&
+      Number.isInteger(value) &&
+      this.checkIfFixable(value, this.fixedInteger, type)
+    ) {
+      this.fixedInteger = value;
+    } else if (
+      type === 'unsignedInt' &&
+      Number.isInteger(value) &&
+      value >= 0 &&
+      this.checkIfFixable(value, this.fixedUnsignedInt, type)
+    ) {
+      this.fixedUnsignedInt = value;
+    } else if (
+      type === 'positiveInt' &&
+      Number.isInteger(value) &&
+      value > 0 &&
+      this.checkIfFixable(value, this.fixedPositiveInt, type)
+    ) {
+      this.fixedPositiveInt = value;
+    } else {
+      throw new MismatchedTypeError('number', value, type);
+    }
+  }
+
+  /**
+   * Fixes a string to this element.
+   * @see {@link fixValue}
+   * @param {string} value - the string value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {TypeNotFoundError} when the value does not match the type of the ElementDefinition
+   */
+  fixString(value: string): void {
+    if (!this.hasSingleType()) {
+      throw new NoSingleTypeError(typeof value);
+    }
+    const type = this.type[0].code;
+    if (type === 'string' && this.checkIfFixable(value, this.fixedString, type)) {
+      this.fixedString = value;
+    } else if (
+      type === 'uri' &&
+      /^\S*$/.test(value) &&
```

Comment:
Assuming you got these regular expressions from the FHIR spec, I think it would be good to include a link to the spec where the regular expressions are specified.  You probably don't need a link for every regex -- but putting a link at the top of the function might be good.